.github/workflows/build_and_test.yml

```diff
@@ -1,39 +1,79 @@
 # This workflow will build a Java project with Maven
 # For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-maven
 
-name: Java CI with Maven
+name: Build
 
 on:
   push:
 
+env: 
+  IS_NON_FEATURE_BRANCH: ${{ github.ref == 'refs/heads/develop' || github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/heads/release') }}
+
 jobs:
   build:
     runs-on: ubuntu-latest
     strategy:
       matrix:
         # test against latest update of each major Java version, as well as specific updates of LTS versions:
         java: [8, 11.0.x, 13 ]
-
+    
     steps:
-    - uses: actions/checkout@v2
-      
-    - uses: actions/cache@v1
+    - name: Checkout the code
+      uses: actions/checkout@v2
+
+    - name: Cache the maven artifacts
+      uses: actions/cache@v1
       with:
         path: ~/.m2/repository
         key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
         restore-keys: |
           ${{ runner.os }}-maven-
           
-    - uses: actions/setup-java@v1
+    - name: Configure java
+      uses: actions/setup-java@v1
       with:
         java-version: ${{ matrix.java }}
 
-    - name: Build with Maven
-      run: mvn -B package --file pom.xml
+    - name: Build and run tests
+      run: mvn -B package 
+
+    - run: echo ${{ env.IS_NON_FEATURE_BRANCH }}
+
+    - name: Run integration tests
+      run: mvn verify
+      if: env.IS_NON_FEATURE_BRANCH == 'true'
+      env:
+        REVAI_ACCESS_TOKEN: ${{ secrets.INTEGRATION_TEST_ACCESS_TOKEN }}
     
-    - run: mkdir staging && cp target/*.jar staging
+    - name: Save build artifacts to staging area
+      run: mkdir staging && cp target/*.jar staging
     
-    - uses: actions/upload-artifact@v1
+    - name: Upload build artifacts
+      uses: actions/upload-artifact@v1
       with:
         name: Package
         path: staging
+    
+    - name: Notify slack fail
+      if: failure() && env.IS_NON_FEATURE_BRANCH == 'true'
+      env:
+        SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
+      uses: voxmedia/github-action-slack-notify-build@v1
+      with:
+        channel: revai-alerts-prod
+        status: FAILED
+        color: danger
+
+  slack_success:
```

Comment:
i have the failure notification as a step in the build job, but success notification as a separate job, this way it only gets sent once instead of 3 times (because build job uses matrix strategy to test with multiple java versions).

it does mean failure would likely get sent multiple times because probably most errors would happen on all java versions. we can probably fix it but it is a bit tricky so I am deferring it for a future issue, hopefully build failures are not that frequent.